Src/FluentAssertions/Common/ExpressionExtensions.cs

```diff
@@ -95,21 +96,41 @@ public static MemberPath GetMemberPath<TDeclaringType, TPropertyType>(
                     node = methodCallExpression.Object;
                     segments.Add($"[{argumentExpression.Value}]");
                     break;
+                case ExpressionType.New:
+                    var newExpression = (NewExpression)node;
+
+                    foreach (var member in newExpression.Arguments)
+                    {
+                        selectors.Add(string.Join(".", member.ToString().Split('.').Skip(1)));
+                        declaringTypes.Add(((MemberExpression)member).Member.DeclaringType);
+                    }
+
+                    node = null;
+                    break;
 
                 default:
                     throw new ArgumentException(GetUnsupportedExpressionMessage(expression.Body), nameof(expression));
             }
         }
 
-        // If any members were accessed in the expression, the first one found is the last member.
         Type declaringType = declaringTypes.FirstOrDefault() ?? typeof(TDeclaringType);
 
-        IEnumerable<string> reversedSegments = segments.AsEnumerable().Reverse();
-        string segmentPath = string.Join(".", reversedSegments);
+        if (selectors.Count == 0)
```

Comment:
It is necessary to use a new `List` which stores multiple complete paths instead of n tokens (which results in one path).

An alternative would be to use e.g. a `Dictionary<int, List<string>>`, but I don't think this makes much better here...